scripts/build_renewable_profiles.py

```diff
@@ -180,220 +179,158 @@
-    for i in regions.index:
-        row = layoutmatrix[i]
-        distances = haversine(regions.loc[i, ['x', 'y']], cell_coords[row.indices])[0]
-        average_distance.append((distances * (row.data / row.data.sum())).sum())
+    centre_of_mass = []
+    for bus in buses:
+        row = layoutmatrix.sel(bus=bus).data
+        nz_b = row != 0
+        row = row[nz_b]
+        co = coords[nz_b]
+        distances = haversine(bus_coords.loc[bus],  co)
+        average_distance.append((distances * (row / row.sum())).sum())
+        centre_of_mass.append(co.values.T @ (row / row.sum()))
 
     average_distance = xr.DataArray(average_distance, [buses])
+    centre_of_mass = xr.DataArray(centre_of_mass, [buses, ('spatial', ['x', 'y'])])
+
 
     ds = xr.merge([(correction_factor * profile).rename('profile'),
-                   capacities.rename('weight'),
-                   p_nom_max.rename('p_nom_max'),
-                   layout.rename('potential'),
-                   average_distance.rename('average_distance')])
+                    capacities.rename('weight'),
+                    p_nom_max.rename('p_nom_max'),
+                    potential.rename('potential'),
+                    average_distance.rename('average_distance')])
 
-    if snakemake.wildcards.technology.startswith("offwind"):
-        import geopandas as gpd
-        from shapely.geometry import LineString
 
-        offshore_shape = gpd.read_file(snakemake.input.offshore_shapes).unary_union
+    if snakemake.wildcards.technology.startswith("offwind"):
+        logger.info('Calculate underwater fraction of connections.')
+        offshore_shape = gpd.read_file(paths['offshore_shapes']).unary_union
         underwater_fraction = []
-        for i in regions.index:
-            row = layoutmatrix[i]
-            centre_of_mass = (cell_coords[row.indices] * (row.data / row.data.sum())[:,np.newaxis]).sum(axis=0)
-            line = LineString([centre_of_mass, regions.loc[i, ['x', 'y']]])
-            underwater_fraction.append(line.intersection(offshore_shape).length / line.length)
+        for bus in buses:
+            p = centre_of_mass.sel(bus=bus).data
+            line = LineString([p, regions.loc[bus, ['x', 'y']]])
+            frac = line.intersection(offshore_shape).length/line.length
+            underwater_fraction.append(frac)
 
         ds['underwater_fraction'] = xr.DataArray(underwater_fraction, [buses])
 
     # select only buses with some capacity and minimal capacity factor
     ds = ds.sel(bus=((ds['profile'].mean('time') > config.get('min_p_max_pu', 0.)) &
-                     (ds['p_nom_max'] > config.get('min_p_nom_max', 0.))))
+                      (ds['p_nom_max'] > config.get('min_p_nom_max', 0.))))
 
     if 'clip_p_max_pu' in config:
-        ds['profile'].values[ds['profile'].values < config['clip_p_max_pu']] = 0.
+        min_p_max_pu = config['clip_p_max_pu']
+        ds['profile'] = ds['profile'].where(ds['profile'] >= min_p_max_pu, 0)
```

Comment:
```suggestion
        ds['profile'] = ds['profile'].where(ds['profile'] >= min_p_max_pu, min_p_max_pu)
```

Why not use `ds.clip(...)` here?

Reply:
again, it's set to zero :)